File: src/main/java/moe/sota/decompiler/services/TypeService.java
package moe.sota.decompiler.services;

import java.util.List;
import lombok.experimental.UtilityClass;
import moe.sota.decompiler.models.FileModel;
import moe.sota.decompiler.types.ClassType;
import moe.sota.decompiler.types.ImageType;
import moe.sota.decompiler.types.ManifestType;
import moe.sota.decompiler.types.Type;
import org.jetbrains.annotations.Nullable;

@UtilityClass
public class TypeService {
    private static final List<Type> TYPES = List.of(new ClassType(), new ImageType(), new ManifestType());

    public static @Nullable Type getType(FileModel fileModel) {
        for (Type format : TYPES) if (format.isFormat(fileModel)) return format;
        return null;
    }
}

File: src/main/kotlin/moe/sota/decompiler/services/TypeService.kt
package moe.sota.decompiler.services

import moe.sota.decompiler.models.FileModel
import moe.sota.decompiler.types.ClassType
import moe.sota.decompiler.types.ImageType
import moe.sota.decompiler.types.ManifestType
import moe.sota.decompiler.types.Type

object TypeService {
    private val types = listOf(ClassType, ImageType, ManifestType)

    fun getType(fileModel: FileModel): Type? = types.firstOrNull { it.isFormat(fileModel) }
}
